pkg/api/types.go

```diff
@@ -2154,3 +2154,32 @@ type RangeAllocation struct {
 	// a single allocated address (the fifth bit on CIDR 10.0.0.0/8 is 10.0.0.4).
 	Data []byte `json:"data"`
 }
+
+// LockSpec is the specification of a resource lock.
+type LockSpec struct {
```

Comment:
So are things fine as they are then?  Leaving HeldBy in the lock impl as a string?  That field is actually used by the update method so removing it would leave the lock open to be stolen by anyone.  Granted it's not the most secure method since anyone can spoof the data if they want, but they have to have access to the system and at that point they can do just about anything anyway. :)